tests/test_logging.py

```
@@ -0,0 +1,318 @@
+    sync-route threadpool), so the value propagates back to the middleware.
+    """
+    from fastapi import Depends
+
+    app = FastAPI()
+    app.add_middleware(AccessLogMiddleware)
+
+    async def fake_auth() -> None:
+        user_id_var.set("user-77")
+
+    @app.get("/whoami")
+    def whoami(_=Depends(fake_auth)):
```

Ruff (0.15.13)

[warning] Missing return type annotation for private function `whoami`

(ANN202)